File: frontend/types/meeting.ts
export interface User {
  id: number;
  name: string;
  email: string;
  created_at: string;
}

export interface Meeting {
  id: number;
  meeting_code: string;
  title: string;
  description: string | null;
  scheduled_at: string | null;
  duration: number;
  meeting_type: "instant" | "scheduled";
  status: "scheduled" | "active" | "ended";
  created_at: string;
  invite_link: string;
  host: User;
}

export interface ScheduleMeetingRequest {
  title: string;
  description?: string;
  scheduled_at: string;
  duration: number;
}

File: backend/app/models/meeting.py
from datetime import datetime, timezone
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text
from sqlalchemy.orm import relationship
from app.database import Base


class Meeting(Base):
    __tablename__ = "meetings"

    id = Column(Integer, primary_key=True, index=True)
    meeting_code = Column(String, unique=True, nullable=False, index=True)
    title = Column(String, nullable=False)
    description = Column(Text, nullable=True)
    scheduled_at = Column(DateTime, nullable=True)
    duration = Column(Integer, default=60)  # duration in minutes
    meeting_type = Column(String, nullable=False)  # "instant" or "scheduled"
    status = Column(String, nullable=False, default="scheduled")  # "scheduled", "active", "ended"
    host_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    # Relationship back to the host user
    host = relationship("User", back_populates="hosted_meetings")

    # A meeting can have many participants
    participants = relationship("Participant", back_populates="meeting", cascade="all, delete-orphan")
